### PR TITLE
 🐛 Fix race condition when opening/closing sidebar.

### LINKED ISSUE #23572: Cannot read property 'hiddenElementInfos' of undefined
```
Error: Cannot read property 'hiddenElementInfos' of undefined
at (https://raw.githubusercontent.com/ampproject/amphtml/1907231757120/src/modal.js:228)
at setModalAsClosed (https://raw.githubusercontent.com/ampproject/amphtml/1907231757120/extensions/amp-sidebar/0.1/amp-sidebar.js:385)
at (https://raw.githubusercontent.com/ampproject/amphtml/1907231757120/src/service/resources-impl.js:955)
at (https://raw.githubusercontent.com/ampproject/amphtml/1907231757120/src/service/vsync-impl.js:460)
at callTask_ (https://raw.githubusercontent.com/ampproject/amphtml/1907231757120/src/service/vsync-impl.js:417)
```

It looks like `setModalAsClosed` can get called when the `modalEntryStack` does not have any elements to pop.

go/ampe/CNmerbKst4CwWw

### PR DESCRIPTION
**Pull request by sparhami**
- Keep track of the opened state as a variable and not an attribute, so
it is updated immediately. This makes sure multiple open/close calls in
a row do not cause issues for the modal or history stacks.

Fixes #23572

### CODE REVIEW

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 1]
```diff
@@ -153,7 +156,7 @@ export class AmpSidebar extends AMP.BaseElement {
       this.fixIosElasticScrollLeak_();
     }
 
-    if (this.isOpen_()) {
+    if (element.hasAttribute('open')) {
```
Comment (cathyxz) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Do you also need to set `this.opened_ = true` either here or in `open_()`? I don't see the change where the initial state from the `open` attribute sets the `this.opened_` variable to true.

File: extensions/amp-sidebar/0.1/test/test-amp-sidebar.js [hunk 2]
```diff
@@ -379,27 +399,27 @@ describes.realWin(
           owners.schedulePause = sandbox.spy();
           owners.scheduleResume = sandbox.spy();
 
-          expect(impl.isOpen_()).to.be.false;
+          expect(sidebarElement.hasAttribute('open')).to.be.false;
```
Comment (cathyxz) [inline extensions/amp-sidebar/0.1/test/test-amp-sidebar.js]: Do these tests still test what they're supposed to test now that we don't use this attribute to guard whether or not the sidebar is `open`?

Comment (sparhami) [reply extensions/amp-sidebar/0.1/test/test-amp-sidebar.js]: I think they still check the attribute is set, which can be used for styling.

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 1 above)
Comment (sparhami) [reply extensions/amp-sidebar/0.1/amp-sidebar.js]: As discussed off-line, this doesn't seem to have worked for a long time and never validated.

**Pull request merged**